### ISSUE TITLE
Unstable representation of self variable

### ISSUE THREAD

**Issue by gkossakowski** [sbt/sbt#2504]
Project with three source files:

``` scala
// Global.scala
class Global { 
  class Foo extends Namers
}

// Namers.scala
class Namers {
  class Namer { thisNamer => }
}

// Client.scala
class Client(global: Global)
```
## steps
1. compile
2. add a whitespace to `Global.scala` and watch `Client.scala` being recompiled
## problem

Whitespace change causes recompilation of dependent files.
## notes

The problem is unstable representation of classes that have a self variable (check `Namer`). This instability comes from not forcing info of the symbol that represent self type/self variable. This is a regression caused by 1ce11230546639fd4a06ab96d19784979a7a358c.

**Comment by gkossakowski** [comment]
I discovered this issue while testing incremental compilation on `scala/scala` repo. We've debugged it together with @adriaanm and the fix is on its way.

**Comment by adriaanm** [comment]
See https://github.com/adriaanm/scala/commit/d6f58d820778a12cc3eda13872640626fd5c53a0

**Comment by dwijnand** [comment]
Is this a blocker for 0.13.12?

**Comment by eed3si9n** [comment]
@dwijnand Fixed in 0.13.12, but not sure if it's forward ported.

**Comment by dwijnand** [comment]
Ok cool. Forward ported in sbt/zinc#90, so we're done here.
